### PR TITLE
Add DH_get_2048_256 to compatibility layer.

### PR DESCRIPTION
**Pull request by haydenroche5**
# Description

This commit adds `wolfSSL_DH_get_2048_256`.

# Testing

Added a unit test and verified that the output is the same between the wolfSSL and OpenSSL versions of the function.

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -34107,6 +34107,121 @@ int wolfSSL_DH_set0_key(WOLFSSL_DH *dh, WOLFSSL_BIGNUM *pub_key,
     return SetDhInternal(dh);
 }
 
+WOLFSSL_DH* wolfSSL_DH_get_2048_256(void)
```
Comment (SparkiDev) [inline src/ssl.c:34110]: Where did these parameters come from?
(standard)

Comment (haydenroche5) [reply src/ssl.c:34110]: RFC 5114

Comment (SparkiDev) [reply src/ssl.c:34110]: Please put this into the code.
Thanks.

Comment (haydenroche5) [reply src/ssl.c:34110]: Added a comment.

File: src/ssl.c [hunk 2]
```diff
@@ -34107,6 +34107,123 @@ int wolfSSL_DH_set0_key(WOLFSSL_DH *dh, WOLFSSL_BIGNUM *pub_key,
...
+    }
+    if (err == 0) {
+        qBn = wolfSSL_BN_bin2bn(qHex, sizeof(qHex), NULL);
+        if (qBn == NULL) {
+            WOLFSSL_MSG("Error converting q hex to WOLFSSL_BIGNUM.");
+            err = 1;
+        }
+    }   
+    if (err == 0) {
+    #if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L
```
Comment (dgarske) [inline src/ssl.c:34199]: If you rebase this PR to latest master the logic here is now `#if defined(OPENSSL_ALL) || \
    defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L`

**Review by dgarske — CHANGES_REQUESTED** [review]

**Review by dgarske — APPROVED** [review]
Can you please note in the PR comments which OSP project this is being added for? Some users may not need this function... did you consider wrapping it with something other than `OPENSSL_EXTRA`? Also appears to be deprecated in v3.0 openssl...

File: src/ssl.c [hunk 3]
```diff
@@ -34128,6 +34128,124 @@ int wolfSSL_DH_set0_key(WOLFSSL_DH *dh, WOLFSSL_BIGNUM *pub_key,
...
+    WOLFSSL_BIGNUM* pBn = NULL;
+    WOLFSSL_BIGNUM* gBn = NULL;
+    WOLFSSL_BIGNUM* qBn = NULL;
+
+    ret = wolfSSL_DH_new();
+    if (ret == NULL) {
+        err = 1;
+    }
+    if (err == 0) {
+        pBn = wolfSSL_BN_bin2bn(pHex, sizeof(pHex), NULL);
```
Comment (dgarske) [inline src/ssl.c:34199]: Looks like you'll need to add a cast to `int` for the `sizeof()` on these three. `wolfSSL_BN_bin2bn` takes `int` for `len`.

**Review by dgarske — CHANGES_REQUESTED** [review]

**Comment by dgarske** [comment]
> > Can you please note in the PR comments which OSP project this is being added for?
> 
> This is for OpenSplice, but according to @cconlon the code we're working with is not the same as the open source version, so I'm not sure we'll add an OpenSplice macro.
> 
> > Some users may not need this function... did you consider wrapping it with something other than `OPENSSL_EXTRA`?
> 
> Generally, I've preferred to add any compatibility layer functions to `OPENSSL_EXTRA` rather than gating on a more specific open source project macro. My reasoning is that users of the compatibility layer are less likely to care about footprint than our resource-constrained/embedded users. Given that, it seems preferable to me to have more of the OpenSSL API available by default rather than requiring further configure flags.
> 
> > Also appears to be deprecated in v3.0 openssl...
> 
> Is there something specific you want me to do here for that? I've not seen us do anything w.r.t. OpenSSL's deprecations in the compatibility layer before.

Thanks, no actions needed on this. I did post another review on the sizeof() that will need a cast.

File: src/ssl.c (hunk 3 above)
Comment (haydenroche5) [reply src/ssl.c:34199]: Thanks. Fixed.

**Review by dgarske — APPROVED** [review]

**Pull request merged**